src/EditorFeatures/CSharpTest/EditAndContinue/TopLevelEditingTests.cs

```diff
@@ -18540,5 +18540,228 @@ public void TopLevelStatements_Reorder()
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.Delete, "partial " + keyword + " C", FeaturesResources.method));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void PrimaryConstructors_13([CombinatorialValues("class", "struct")] string keyword)
+        {
+            var src1 = "partial " + keyword + " C(); partial " + keyword + " C { }";
+            var src2 = "partial " + keyword + " C(); partial " + keyword + " C { void M() { } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.Insert, "void M()", FeaturesResources.method));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void PrimaryConstructors_14([CombinatorialValues("class", "struct")] string keyword)
+        {
+            var src1 = "partial " + keyword + " C(); partial " + keyword + " C { void M() { } }";
+            var src2 = "partial " + keyword + " C(); partial " + keyword + " C { void M() { ToString(); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.Update, "void M()", FeaturesResources.method));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void PrimaryConstructors_15([CombinatorialValues("class", "struct")] string keyword)
+        {
+            var src1 = "partial " + keyword + " C(); partial " + keyword + " C { void M() { } }";
+            var src2 = "partial " + keyword + " C {} partial " + keyword + " C { void M() { ToString(); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.Update, "void M()", FeaturesResources.method));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void PrimaryConstructors_16([CombinatorialValues("class", "struct")] string keyword)
+        {
+            var src1 = "partial " + keyword + " C {} partial " + keyword + " C { void M() { } }";
+            var src2 = "partial " + keyword + " C(); partial " + keyword + " C { void M() { ToString(); } }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.Update, "void M()", FeaturesResources.method));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void PrimaryConstructors_17([CombinatorialValues("class", "struct")] string keyword)
+        {
+            var src1 = keyword + " C(int a) { }";
+            var src2 = keyword + " C(long a) { }";
+
+            var edits = GetTopEdits(src1, src2);
+
+            edits.VerifySemanticDiagnostics(
+                Diagnostic(RudeEditKind.Update, "long a", FeaturesResources.parameter));
+        }
+
+        #endregion
```

Comment:
>#endregion [](http://example.com/codeflow?start=8&length=10)

Consider testing edits of explicit constructors other than the primary constructor, and edits of other members such as instance and static methods and fields.

Reply:
> Consider testing edits of explicit constructors other than the primary constructor, and edits of other members such as instance and static methods and fields.

Testing of field initializers is not particularly interesting at the moment because they do not change capturing of primary constructor parameters. Also, at the moment, I do not care if their edits are flagged as rude edits or not. The goal is to error on the safe side.